src/libmongoc/tests/unified/entity-map.c

```diff
@@ -1119,13 +1151,43 @@ entity_map_add_changestream (entity_map_t *em,
       em, id, "changestream", (void *) changestream, error);
 }
 
+void
+entity_findcursor_iterate_until_document_or_error (
+   entity_findcursor_t *findcursor,
+   const bson_t **document,
+   bson_error_t *error,
+   const bson_t **error_document)
+{
+   *document = NULL;
+
+   if (findcursor->first_result) {
+      *document = findcursor->first_result;
+      findcursor->first_result = NULL;
+      return;
+   }
+
+   while (!mongoc_cursor_next (findcursor->cursor, document)) {
+      if (mongoc_cursor_error_document (
+             findcursor->cursor, error, error_document)) {
+         return;
+      }
+   }
+}
+
 bool
-entity_map_add_cursor (entity_map_t *em,
-                       const char *id,
-                       mongoc_cursor_t *cursor,
-                       bson_error_t *error)
+entity_map_add_findcursor (entity_map_t *em,
+                           const char *id,
+                           mongoc_cursor_t *cursor,
+                           const bson_t *first_result,
+                           bson_error_t *error)
 {
-   return _entity_map_add (em, id, "cursor", (void *) cursor, error);
+   entity_findcursor_t *findcursor;
+
+   findcursor =
```

Comment:
Ok, this bason_malloc() abort()s the process on failure.

Reply:
Correct.